master-firmware/src/strategy.cpp

```diff
@@ -348,28 +348,45 @@ struct PickupBlocks : actions::PickupBlocks {
     {
         const int x_mm = state.blocks_pos[blocks_id][0];
         const int y_mm = state.blocks_pos[blocks_id][1];
-        NOTICE("Picking up some blocks at %d %d", x_mm, y_mm);
+        NOTICE("Picking up some blocks at %d %d", MIRROR_X(m_color, x_mm), y_mm);
 
         enum lever_side_t lever_side = LEVER_SIDE_LEFT;
         lever_t* lever = MIRROR_LEFT_LEVER(m_color);
-        int a_deg = -45;
+        int offset_a_deg = 0;
 
         if (state.lever_full_left) {
             lever_side = LEVER_SIDE_RIGHT;
             lever = MIRROR_RIGHT_LEVER(m_color);
-            a_deg += 180;
+            offset_a_deg = 180;
         }
 
-        se2_t blocks_pose = se2_create_xya(MIRROR_X(m_color, x_mm), y_mm, 0);
+        se2_t cubes_pose = se2_create_xya(MIRROR_X(m_color, x_mm), y_mm, 0);
+        se2_t pickup_poses[4] = {
+            se2_create_xya(MIRROR_X(m_color, x_mm - 160), y_mm - 160, MIRROR_A(m_color, -45)),
+            se2_create_xya(MIRROR_X(m_color, x_mm + 160), y_mm - 160, MIRROR_A(m_color,  45)),
+            se2_create_xya(MIRROR_X(m_color, x_mm + 160), y_mm + 160, MIRROR_A(m_color, 135)),
+            se2_create_xya(MIRROR_X(m_color, x_mm - 160), y_mm + 160, MIRROR_A(m_color, 225)),
+        };
+        int num_possible_poses = sizeof(pickup_poses) / sizeof(pickup_poses[0]);
```

Comment:
[nitpick] Since we are doing c++, we might use std::array here instead of the sizeof hack :wink:

Reply:
good point